### PR TITLE
Allow w:h entry in bauhaus slider popup, direct user there for custom aspect ratios in Framing module

### PR DESCRIPTION
**Pull request by ralfbrown**
Although setting the dropdown to "non-editable" doesn't prevent the user from typing arbitrary text into the dropdown menu, it won't be displayed once the menu collapses.  Update the tooltips for both the dropdown and slider to direct the user to right-clicking on the slider and entering w:h there for custom aspect ratios.

### CODE REVIEW

**Review by TurboGit — CHANGES_REQUESTED** [review]
Don't seem to work as expected. Enter for example 4:5 and the aspect ratio is set to 1.0 (square). It seems that it tries to match the known values in the combo.

**Comment by ralfbrown** [comment]
Hmm, 7:6 worked for me, and that's not one of the predefined ratios.  But it occurs to me that the slider has a minimum value of 1.0 and 4:5 would give 0.8 and be clamped.  So we'll need code to take the reciprocal when the user sets a value < 1 (either by ratio or decimal number).

**Comment by dterrahe** [comment]
> So we'll need code to take the reciprocal when the user sets a value < 1

This could be added in `gui_changed`. You may also want to flip `orientation` _landscape_ <-> _portrait_.
One hitch of course is that the hard range of the slider would be enforced before you even get there, so the hard min would have to be set (in `gui_init`) at (almost) zero and the active min would need to be reset to 1 again each time after taking the reciprocal.

**Comment by ralfbrown** [comment]
I figured hard min would be an issue, and flipping the orientation had occurred to me.  The question is, how exactly to handle that - would 4:3 always force landscape and 3:4 always portrait, or would 4:3 keep and 3:4 toggle?  (I lean to the former.)

**Comment by dterrahe** [comment]
> would 4:3 always force landscape and 3:4 always portrait

That would be significant departure from how it works currently and I'm not sure how it would work; I type in 3:4 and it flips to portrait _and_ calculates reciprocal? But now, if I just retype the 4:3 it is showing it flips back to landscape?

Or would 3:4 just be kept (i.e. slider range extends <1 permanently ("soft min = hard min = 1/3"). But now the dropdown also needs to list both 3:4 as well as 4:3. And the separate orientation dropdown is superfluous.

In other words; I lean to the latter ;-)

**Comment by ralfbrown** [comment]
Point taken, hadn't thought of those aspects.

**Comment by dterrahe** [comment]
Oops! I accidentally pushed a change directly to this PR. I didn't know I could do that! Assumed that it was in your repo @ralfbrown and that github would require authentication. I was going to submit a PR to you. Hope you don't mind/that it doesn't mess things up if you make any more changes.

Anyway; this deals with <1 aspect ratios by transposing them to 1-3 range and flipping the orientation (if it is not "auto").

Small complications from special values "image" and "constant border" being 0 and -1 (i.e. being floored at hard-min = 1/3 rather than 1 as before) have been addressed.

**Comment by ralfbrown** [comment]
There's an option (set by default) when opening a PR to let maintainers push to the PR.

Thanks!  I'll need to double-check one concern I had thought of which I don't see addressed in your update - when scrolling the slider past the soft minimum, will the orientation flip-flop with every update?

**Comment by dterrahe** [comment]
Alternatively, we could rethink this whole PR:
- leave slider range as 1-3
- don't change tooltips
- continue to allow freeform entry in combo
- if entry not recognized (pos = -1), use `dt_calculator_solve` to get the value of string; if <1, take reciprocal and flip orientation
- support ":" in `dt_calculator_solve (as per this PR).
- set (calculated) value in slider. It will update combo (to "custom..." if needed).

**Comment by ralfbrown** [comment]
Since my last post here, I had the thought of having the ':' in the calculator being interpreted as a ratio which is always >= 1 rather than a strict alias of division.  We'd lose the automatic orientation flip, but it would eliminate all the other issues arising from trying to special-case an aspect ratio < 1.

**Comment by ralfbrown** [comment]
And implemented...
I'll close this PR in favor of the new #14461.
